### ISSUE TITLE
Poor Quality?

### ISSUE THREAD

**Issue by LynnHo** [taki0112/SphereGAN-Tensorflow#1]
I have also implemented a SphereGAN, but the quality is not as good as the paper claimed. The results of your code seem also not good enough. Do you have any idea?

**Comment by taki0112** [comment]
For the paper, only 32x32.
I think my image quality is poor because it is 64x64 image.

**Comment by LynnHo** [comment]
@taki0112 Oh thanks a lot！ By the way, I saw you post many excellent implementations of GAN works. So I would like to ask you, which GAN technique is more preferable in your work. I have tried many GAN variants, but the most practical ones are still WGAN-GP(LP) and DRAGAN, or maybe SNGAN.

**Comment by taki0112** [comment]
I have a similar opinion with you.
I think the following things are useful.

1. [Simple-gradient penalty](https://github.com/taki0112/Tensorflow-Cookbook/blob/master/ops.py#L1023)  To stable in training GAN, It is used in StyleGAN.

2. [Spectral normalization](https://github.com/taki0112/Spectral_Normalization-Tensorflow)

3. LSGAN in Image-to-Image Translation tasks.

**Comment by LynnHo** [comment]
Thanks a lot!
